apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,429 @@
+import { EditSection } from "../../../shared";
+import "./sms-recovery.scss";
+
+/**
+ * SMS key.
+ */
+const SMS: string = "sms";
+
+/**
+ * Prop types for the SMSRecoveryComponent component.
+ * Also see {@link SMSRecovery.defaultProps}
+ */
+interface SMSRecoveryProps extends IdentifiableComponentInterface {
+    onAlertFired: (alert: AlertInterface) => void;
+    enableEditMobile?: boolean;
+}
+
+/**
+ * SMS recovery section.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const SMSRecovery: React.FunctionComponent<SMSRecoveryProps> = (
+    props: SMSRecoveryProps
+): ReactElement => {
+
+    const {
+        onAlertFired,
+        enableEditMobile,
+        ["data-componentid"]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
+
+    const profileInfo: BasicProfileInterface = useSelector(
+        (state: AppState) => state.authenticationInformation.profileInfo
+    );
+    const phoneNumberSchema: ProfileSchema = useSelector((state: AppState) => {
+        const phoneNumberSchemas: ProfileSchema = state.authenticationInformation.profileSchemas.find(
+            (profileSchema: ProfileSchema) => profileSchema.name === "phoneNumbers");
+
+        if (phoneNumberSchemas && phoneNumberSchemas.subAttributes) {
+            const mobileSchema: ProfileSchema = phoneNumberSchemas.subAttributes?.find(
+                (subAttribute: ProfileSchema) => subAttribute.name === "mobile");
+
+            if (mobileSchema) return mobileSchema;
+        }
+
+        return phoneNumberSchemas;
+    });
+    const activeForm: string = useSelector((state: AppState) => state.global.activeForm);
+
+    const [ mobile, setMobile ] = useState<string>(undefined);
+    const [ editedMobile, setEditedMobile ] = useState<string>(undefined);
+
+    let mobileType: string;
+
+    useEffect(() => {
+        if (isEmpty(profileInfo)) {
+            dispatch(getProfileInformation());
+        }
+    }, []);
+
+    const handleUpdate = (mobile: string) => {
+        const data: {
+            Operations: {
+                op: string;
+                value: Record<string, unknown>;
+            }[];
+            schemas: string[];
+        } = {
+            Operations: [
+                {
+                    op: "replace",
+                    value: {}
+                }
+            ],
+            schemas: [ "urn:ietf:params:scim:api:messages:2.0:PatchOp" ]
+        };
+
+        data.Operations[0].value = {
+            phoneNumbers: mobileType || phoneNumberSchema
+                ? [
+                    {
+                        type: mobileType || phoneNumberSchema.name,
+                        value: mobile
+                    }
+                ]
+                : [ mobile ]
+        };
+
+        updateProfileInfo(data)
+            .then(() => {
```

Comment:
Let's handle the submitting and loading states as well.

Eg: submit buttons need to show loading spinner when the api request is in progress.

Reply:
There are no loading spinners in the current implementation for the email recovery component which is a similar component to this and is in the same place. Adding loaders for only this component will result in an inconsistent UX.

Reply:
Let's add the state to handle submitting UX for this this component.

We will add the necessary improvements for other components from the UX team end :)

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/6418/commits/c7cf2af96a89c36059f36a3408340f64b9cbd14a